haystack/schema.py

```diff
@@ -849,22 +929,70 @@ def wrong_examples(
 
         return []
 
+    def _get_document_relevance_criterion(
+        self,
+        document_scope: Literal[
+            "id", "context", "id_and_context", "id_or_context", "answer", "id_or_answer"
+        ] = "id_or_answer",
+        answer_scope: Literal["any", "context", "document", "document_and_context"] = "any",
+    ) -> Literal[
+        "id",
+        "context",
+        "id_and_context",
+        "id_or_context",
+        "answer",
+        "context_and_answer",
+        "id_and_answer",
+        "id_and_context_and_answer",
+        "id_or_answer",
+    ]:
+        """
+        Combines document_scope and answer_scope to create the document_relevance_criterion.
+        """
+        answer_scope_to_doc_relevance_crit = {
+            "context": "context_and_answer",
+            "document": "id_and_answer",
```

Comment:
As context becomes context_and_answer, I think it would be more intuitive to have something like document_id become document_id_and_answer and document_id_and_context become document_id_and_context.